src/AppInstallerCLIE2ETests/WinGetUtil/WinGetUtilSQLiteIndex.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace AppInstallerCLIE2ETests.WinGetUtil
+{
+    using System;
+    using System.IO;
+    using System.Runtime.InteropServices;
+    using NUnit.Framework;
+
+    public class WinGetUtilSQLiteIndex
+    {
+        private string sqlitePath;
+        private readonly uint majorVersion = 1;
+        private readonly uint minorVersion = 2;
+
+        // Manifest example 1
+        private readonly string addManifestsFile_1 = TestCommon.GetTestDataFile(@"WinGetUtil\Manifests\Merged\WinGetUtilTest.Add.yaml");
+        private readonly string updateManifestsFile_1 = TestCommon.GetTestDataFile(@"WinGetUtil\Manifests\Merged\WinGetUtilTest.Update.yaml");
+        private readonly string relativePath_1 = @"manifests\a\AppInstallerTest\WinGetUtilTest\1.0.0.0\WinGetTest.yaml";
+
+        [SetUp]
+        public void SetUp()
+        {
+             this.sqlitePath = TestCommon.GetRandomTestFile(".sql");
+        }
+
+        [Test]
+        public void WinGetUtil_SQLiteIndex_AddManifest()
+        {
+            SQLiteIndex((indexHandle) =>
+            {
+                // Add manifest
+                IntPtr hresult = WinGetUtilWrapper.WinGetSQLiteIndexAddManifest(indexHandle, addManifestsFile_1, relativePath_1);
+                Assert.AreEqual(IntPtr.Zero, hresult);
```

Comment:
>Assert.AreEqual(IntPtr.Zero, hresult); [](http://example.com/codeflow?start=16&length=38)

hresult should be compared to ErrorCode.S_OK(defined in Constants.cs) across all tests (in other files too)

Intptr valus are specifically for native handles

Reply:
I just saw all dllimport functions has preserveSig = false, so we probably don't have to do these checks since if the call failed, they are automatically converted to exceptions.

And the return type in dllimport could be void. I saw definitions in IndexCreationTool can also be changed (not your fault)